p8e-shared/src/main/kotlin/io/provenance/p8e/shared/service/AffiliateService.kt

```diff
@@ -53,17 +58,58 @@ class AffiliateService(
         const val AFFILIATE_ENCRYPTION_KEY_PAIR = "affiliate_encryption_key_pair"
         const val AFFILIATE_BECH32_LOOKUP = "affiliate_bech32_lookup"
         const val PUBLIC_KEY_TO_ADDRESS = "public_key_to_address"
+        const val AFFILIATE_SIGNING_KID = "affiliate_signing_key_id"
+        const val AFFILIATE_ENCRYPTION_PUBLIC_KEY = "affiliate_encryption_public_key"
+    }
+
+    /**
+     *  Return the signer to be used
+     *
+     *  [publicKey] publicKey used to validate
+     *  [signer] The specific signer used to sign contracts
+     */
+    fun getSigner(publicKey: PublicKey): SignerImpl {
+        val affiliateRecord = get(publicKey)
+        return if(affiliateRecord?.keyType == SMARTKEY) {
+            signerFactory.getSigner(SignerFactoryParam.SmartKeyParam(affiliateRecord.signingKeyUuid.toString()))
+        } else {
+            signerFactory.getSigner(
+                SignerFactoryParam.PenParam(
+                    KeyPair(affiliateRecord?.publicKey?.value?.toJavaPublicKey(), affiliateRecord?.privateKey?.toJavaPrivateKey())
```

Comment:
right here would probably be the best place to use `!!` since we know they should exist in the database key type or else we could throw an error saying there's a `DATABASE` affiliate with missing fields. That should help so we don't need to `!!` across other places where the signer is used.